### PR TITLE
Update Readme.md and various fixes

### PR DESCRIPTION
**Pull request by Ebola16**
Pulled from #6039. 

- Mention the purpose of the "Release" and "Debug" solution configurations
- Update -d argument help message (to match #6043)
- Grammar / consistency fixes
- Partially address https://bugs.dolphin-emu.org/issues/10563

### CODE REVIEW

**Comment by Ebola16** [comment]
Also, does anyone know where to find readme.md in Visual Studio? I couldn't find it and made these edits on Github''s website so I couldn't use clang-format.

**Comment by JosJuice** [comment]
> Also, does anyone know where to find readme.md in Visual Studio? I couldn't find it and made these edits on Github''s website so I couldn't use clang-format.

It's not part of the solution, so you won't find it in the solution explorer, but you can open it using e.g. File > Open > File. Opening it in a regular text editor works about as well, though.

**Comment by Ebola16** [comment]
Ok I opened readme.md but got and error while running clang-format:
'3' is an unexpected token. The expected token is "" or '''. Line 2, position 66.

I guess that's because clang-format isn't for markdown files.

File: Readme.md [hunk 1]
```diff
@@ -107,6 +107,14 @@ Android apps are compiled using a build system called Gradle. Dolphin's native c
 however, is compiled using CMake. The Gradle script will attempt to run a CMake build
 automatically while building the Java code.
 
+## Visual Studio "Debug" and "Release" Solution Configurations
```
Comment (lioncash) [inline Readme.md]: I disagree with including this sort of thing as a fully separate section. It is not our job to explain how IDEs function. Especially considering CMake can generate build files for several of them. At most, referencing that the Release configuration should be used in the Building for Windows section should be good enough.

Comment (Ebola16) [reply Readme.md]: That accomplishes the same goal but with less detail so I'm fine with that. Done.

**Comment by Ebola16** [comment]
I clarified the "Release" solution configuration information in readme.md. I believe it will convince the curious users to choose the "Release" solution configuration if they want maximum performance.

I also found some minor grammar errors in the CMake files so I added them on to this PR.

@leoetlino may be interested in these changes too.

File: CMake/CheckAndAddFlag.cmake [hunk 2]
```diff
@@ -3,13 +3,13 @@ include(CheckCXXCompilerFlag)
 
 # check_add_add_flag(<variable> <flag> [DEBUG_ONLY | RELEASE_ONLY])
 #
-# Add a C or C++ compilation flag to the current scope
+# Add C or C++ compilation flags to the current scope
```
Comment (leoetlino) [inline CMake/CheckAndAddFlag.cmake]: Please revert this change. As the name and signature indicate, `check_and_add_flag` is used to add a flag, not several flags.

File: Readme.md [hunk 3]
```diff
@@ -43,6 +43,9 @@ Visual Studio 2017 is a hard requirement. Other compilers might be
 able to build Dolphin on Windows but have not been tested and are not
 recommended to be used. Git and Windows 10 SDK 10.0.15063.0 must be installed when building.
 
+The "Release" solution configuration is designed for end users and includes performance
```
Comment (leoetlino) [inline Readme.md]: Not too convinced about the wording here... Developers also have a legitimate reason to use release builds: it's much faster than debug builds, and sometimes issues only happen in release (optimised) builds.

File: CMake/CheckAndAddFlag.cmake (hunk 2 above)
Comment (Ebola16) [reply CMake/CheckAndAddFlag.cmake]: Done. The multiple examples made it seem like multiple flags could be added.

File: Readme.md (hunk 3 above)
Comment (Ebola16) [reply Readme.md]: I'll reword this.

**Comment by Ebola16** [comment]
I re-added a sentence about the Debug solution configuration although this time it is a part of Building for Windows. I don't know if @lioncash will agree but there are users who want to experiment with building Dolphin in Visual Studio. Although most people building Dolphin will know the difference between the two solution configurations, it is important for everyone to know that Release should be used for playing games and Debug is significantly slower. Also, mentioning one solution configuration without mentioning the other forces readers to make assumptions about the unmentioned one. I'd like to avoid that.

**Comment by Ebola16** [comment]
ping?

**Comment by Ebola16** [comment]
Reverted rename of software renderer's "Bypass XFB" to "Disable XFB" since #5498 removes the option.

**Comment by Ebola16** [comment]
Clang Format suggested a change in UberShaderPixel.cpp so I guess I should ping @stenzek to be safe.

**Comment by stenzek** [comment]
@Ebola16 just ignore that file. IIRC it gives a different result with newer versions of clang-format, which means if you include it in your commit it will fail lint.

**Comment by Ebola16** [comment]
@stenzek Thanks, the clang-format changes have been reverted. I didn't know that newer versions of clang-format would disagree with lint. It'd be nice if the buildbots could auto-trust me so I could figure that out on my own. Regardless, I opened https://bugs.dolphin-emu.org/issues/10579 in case something can be done about clang-format and lint disagreements.

Ready for merge again.

**Comment by Ebola16** [comment]
Found some more grammar / consistency fixes

**Comment by Ebola16** [comment]
Updated SD card checkbox ToolTip to reflect SDSC and SDHC support.

Is there a reason why this isn't merged yet? I'd still like to be auto-trusted by the buildbot so I can quickly resolve clang version conflicts.

**Comment by JosJuice** [comment]
I don't think there's any reason to add that to the SD card tooltip.

**Comment by Ebola16** [comment]
@JosJuice I updated the SD card tooltip since there is a lot of old information in forums that says Dolphin doesn't support SDHC. New users may also be unaware of SDHC compatibility. SDHC can significantly improve the Brawl modding experience as you don't need to worry about the old 2GB SDSC limit. The addition is concise and doesn't look bad visually IMO.

If you still want the tooltip removed I can do that but I think it's worth adding.

**Comment by leoetlino** [comment]
I don't think we should include the info either, because:
* there typically isn't any UI change for fixed/newly-implemented functionality -- that's what the progress reports are for
* it's factually incorrect: SDHC support depends on the running title. In older games, Dolphin still won't support SDHC cards (just like console).

**Comment by Ebola16** [comment]
@leoetlino changing my tooltip change to "Supports SDSC and SDHC if games support them." should take care of your second point.

I still feel that this change is beneficial for the reasons I previously stated as the tooltip adds concise and valuable information that users would need to otherwise find on their own, even if it is included in a progress report.

**Comment by JosJuice** [comment]
If we wanted to be consistent and add this kind of "useful information" in all cases where it could be useful, the GUI would be littered with it. So let's not start adding this kind of information.

**Comment by Ebola16** [comment]
Very well, I've given my thoughts on the issue but I'll drop it as it's deemed undeseriable. I kept the added period at the end of the original tooltip for consistency though. 

Ready to merge again. I'd still like to be auto-trusted by the buildbot so I can quickly resolve clang version conflicts.

**Comment by Ebola16** [comment]
I think I did my first merge conflict resolution properly.

Ready to merge again. I'd still like to be auto-trusted by the buildbot so I can quickly resolve clang version conflicts.

**Comment by leoetlino** [comment]
This needs a rebase (right now you have some extra unwanted merge commits).

>Ready to merge again. I'd still like to be auto-trusted by the buildbot so I can quickly resolve clang version conflicts.

I don't have permission to do that. But I don't think it's that useful anyway for documentation changes.

**Comment by Ebola16** [comment]
@leoetlino Rebased. I realize that you may not have permission to auto-trust me but it'd be nice if someone could for future PRs so I don't need to wait on lint's input.
